server/channels/app/export.go

```diff
@@ -491,6 +504,61 @@ func (a *App) exportAllUsers(ctx request.CTX, job *model.Job, writer io.Writer,
 	return profilePictures, nil
 }
 
+func (a *App) exportAllBots(ctx request.CTX, job *model.Job, writer io.Writer, includeProfilePictures bool) ([]string, *model.AppError) {
+	afterId := strings.Repeat("0", 26)
+	cnt := 0
+	profilePictures := []string{}
+
+	for {
+		bots, err := a.Srv().Store().Bot().GetAllAfter(1000, afterId)
+		if err != nil {
+			return profilePictures, model.NewAppError("exportAllBots", "app.user.get.app_error", nil, "", http.StatusInternalServerError).Wrap(err)
+		}
+
+		if len(bots) == 0 {
+			break
+		}
+		cnt += len(bots)
+		updateJobProgress(ctx.Logger(), a.Srv().Store(), job, "bots_exported", cnt)
+
+		for _, bot := range bots {
+			afterId = bot.UserId
+
+			var ownerUsername string
+			owner, err := a.Srv().Store().User().Get(ctx.Context(), bot.OwnerId)
```

Comment:
Can we use `GetMany` to be more efficient? We can change the `pageSize` to be 100 instead to put a cap on the max number of userIDs passed to `GetMany`.

Reply:
Not all owner IDs are actual users IDs, sometimes those can be a plugin. That's the reason I wanted to make the call separately. Alternatively we can filter out bots owned by plugins (or even not export those) and treat separately. Considering that bots shouldn't be too much (?), simpler code made more sense to me. WDYT?

Reply:
Makes sense. :+1: